compute_sdk/globus_compute_sdk/serialize/facade.py

```diff
@@ -34,34 +38,51 @@ class AllowlistWildcard(str, Enum):
     DATA = "globus_compute_sdk.*Data"
 
 
-DeserializerAllowlist = t.Iterable[
-    t.Union[type[SerializationStrategy], str, AllowlistWildcard]
-]
+DeserializerAllowlist = t.Iterable[t.Union[Strategylike, AllowlistWildcard]]
 
 
-def assert_strategy_type_valid(
-    strategy_type: type[SerializationStrategy], for_code: bool
-) -> None:
-    if strategy_type not in SELECTABLE_STRATEGIES:
-        raise SerdeError(
-            f"{strategy_type.__name__} is not a known serialization strategy"
-            f" (must be one of {SELECTABLE_STRATEGIES})"
-        )
+@dataclass
+class ValidatedStragegylike:
+    type_: type[SerializationStrategy]
+    instance: SerializationStrategy
+    import_path: str
+
+
+def validate_strategylike(
+    value: t.Any, *, for_code: bool | None = None
+) -> ValidatedStragegylike:
+    bad_type_error = SerdeError(
+        f"Invalid strategy-like '{value}'. Expected either a SerializationStrategy"
+        f" instance, a subclass of SerializationStrategy, or an import path to one."
+    )
+
+    if isinstance(value, str):
+        try:
+            mod_name, obj_name = value.rsplit(".", 1)
+            mod = importlib.import_module(mod_name)
+            value = getattr(mod, obj_name)
+        except Exception as e:
+            raise bad_type_error from e
+
+    class_ = value if inspect.isclass(value) else type(value)
+    normalized_path = f"{class_.__module__}.{class_.__name__}"
 
-    if strategy_type.for_code != for_code:
+    if not issubclass(class_, SerializationStrategy):
+        raise bad_type_error
+
+    if for_code is not None and class_.for_code != for_code:
         etype = "code" if for_code else "data"
-        gtype = "code" if strategy_type.for_code else "data"
+        gtype = "code" if class_.for_code else "data"
```

Comment:
Every time I read this, I need to reteach myself what it's purpose is: to have these be opposite values for the following string.  Simple, but the if-else makes my brain struggle.  Not at all a problem, but "while you're here", I offer an alternative route.  Consider:
```python
types = ["code", "data"]
etype = types.pop(for_code)
gtype = types.pop()
```
No inline if-statement, and now it's clear, by data structure manipulation, that they must be different.

Reply:
That's fair, I actually find myself tripping over this a bit as well. What about taking a hint from `serialize`, with something like the following?

```
if for_code:
    etype, gtype = "code", "data"
else:
    etype, gtype = "data", "code"
```

Reply:
Hah. That works.  One line, and an indent level more verbose, but it works.  :wink: